File: Phase - 7 Automation Testing Using Java/Cucumber/reports/src/test/java/pages/LoginPage.java
package pages;

import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class LoginPage {
	
	@FindBy(xpath = "//input[@name = \"username\"]")
	public WebElement username;
	
	@FindBy(xpath = "//input[@name = \"password\"]")
	public WebElement password;
	
	@FindBy(xpath = "//button[text() = \" Login \"]")
	public WebElement loginButton;
}

File: Phase - 12 Playwright Automation/POM/pages/LoginPage.ts
import {Page, Locator} from '@playwright/test'

export class LoginPage
{
    readonly page: Page;
    readonly username: Locator;
    readonly password: Locator;
    readonly loginButton: Locator;
    readonly loginTitle: Locator;
    readonly errorMessage: Locator;

    constructor(page: Page) {
        this.page = page;
        this.username = page.locator("//input[@name = 'username']")
        this.password = page.locator("//input[@name = 'password']")
        this.loginButton = page.locator("//button")
        this.loginTitle = page.locator("//h5")
        this.errorMessage = page.locator("//p[text() = 'Invalid credentials']")
    }

    async navigate() {
        await this.page.goto("https://opensource-demo.orangehrmlive.com/web/index.php/auth/login", 
            {'waitUntil' : 'domcontentloaded'}
        );
    }

    async login(username:string, password:string) {
        await this.username.fill(username);
        await this.password.fill(password);
        await this.loginButton.click();
    }

    async getLoginTitle() {
        return await this.loginTitle.textContent();
    }

    async getErrorMessage() {
        return await this.errorMessage.textContent();
    }

}